packages/dds/sequence/src/plan-for-updating-interval-api.md

```diff
@@ -0,0 +1,87 @@
+# Refactoring/Deprecating Interval-related API
+
+## Refactoring
+
+
+1. Keep the IntervalCollection related contents in the current file, it includes:
+
+- `IntervalCollection`: Collection of intervals that supports addition, modification, removal, and efficient spatial querying.
+
+- `LocalIntervalCollection`
+
+- `IIntervalCollectionEvent`
+
+- `IntervalCollectionIterator`
+
+- All corresponding interfaces and helper functions for `IntervalCollection`
+
+2.  Create a new folder for Intervals-related components
+
+- `intervalInterfaces`: stores the generic interval interface (`IInterval`), `ISerializedInterval`, `ISerializableInterval`
+
+- `Interval`: Implementation of the `ISerializableInterval`, whose endpoints are plain-old numbers
+
+- `SequenceInterval`: Implementation of the `ISerializableInterval`, whose ends are associated with positions in a mutatable sequence.
+
+3. Create a new folder for IntervalIndex-related components
+
+- `intervalIndexInterfaces`: stores the generic intervalIndex interface (`IIntervalIndex`), `IOverlappingIntervalsIndex`, `IStartpointInRangeIndex`, `IEndpointInRangeIndex`, and etc.
+
+- `StartpointInRangeIndex`
+
+- `EndpointInRangeIndex`
+
+- `OverlappingIntervalsIndex`
+
+- `OverlappingSequenceIntervalsIndex`
+
+- `SequenceIntervalIndexes`: store the namespace contains specialiazations of indexes which support spatial queries specifically for `SequenceInterval`.
+
+4. Move `IntervalCollectionFactory` and `SequenceIntervalCollectionFactory` to separate files, which are responsible for creating instances of various types of IntervalCollection
```

Comment:
i think you can ditch these two and just have `index.ts` export the creation functions from each intervalIndex file.

I conceptually like the idea of having an entrypoint for all the factories we support, but I'm not sure it tree-shakes well. There might not be harm in having both, but IMO this is the kind of thing I'd rather add if people complain than do proactively.